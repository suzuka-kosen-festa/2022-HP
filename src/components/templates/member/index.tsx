import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import type { MemberProperties } from "./types/model";
import { Title } from "../../molecules/title";
import { MemberCard } from "../../molecules/member-card";
import { Button } from "../../atoms/button";

const Main = tw.main`pt-1 pb-8 sm:(pt-6 pb-40)`;

const MemberBox = tw.div`flex flex-col justify-center items-center`;

const Member: FC<MemberProperties> = ({ rikuto, shoma, members }) => (
  <Main aria-label="member page">
    <Title css={tw`mb-5.5 sm:mb-14`} colorKey="blue">
      Members
    </Title>
    <MemberBox>
      <h2 css={tw`font-zen font-bold text-text text-base mb-9 sm:(text-2xl mb-50)`}>
        第57回鈴鹿高専祭 Re:ROADの実行委員をご紹介します！
      </h2>
      <div css={tw`grid grid-cols-2 gap-8 mb-10 sm:(gap-40 mb-16)`}>
        <MemberCard imagePath={rikuto.imagePath} name={rikuto.name} position="leader" />
        <MemberCard imagePath={shoma.imagePath} name={shoma.name} position="sub" />
      </div>
      <div css={tw`grid grid-cols-3 gap-2 mb-8 sm:mb-40 lg:(grid-cols-5 gap-8)`}>
        {members.map(member => (
          <MemberCard imagePath={member.imagePath} name={member.name} key={member.name} />
        ))}
      </div>
      <Link to="/">
        <Button textStyles={tw`text-white`} colorKey="blue" as="a" role="button">
          トップページに戻る
        </Button>
      </Link>
    </MemberBox>
  </Main>
);

export { Member };
