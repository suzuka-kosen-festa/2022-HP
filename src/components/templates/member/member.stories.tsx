import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { Member } from ".";

type T = typeof Member;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  rikuto: {
    imagePath: "/statics/members/rikuto_kuwahara.webp",
    name: "Rikuto Kuwahara",
  },
  shoma: {
    imagePath: "/statics/members/shoma_kobayashi.webp",
    name: "Shoma Kobayashi",
    position: "sub",
  },
  members: [
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
    {
      imagePath: "/statics/members/rintaro_itokawa.webp",
      name: "Rintaro Itokawa",
    },
  ],
};

export default {
  args,
  argTypes: {
    rikuto: {
      description: "Rikuto Kuwahara's data",
      control: {
        type: "object",
      },
    },
    shoma: {
      description: "Shoma Kobayashi's data",
      control: {
        type: "object",
      },
    },
    members: {
      description: "Other member's data",
      control: {
        type: "array",
      },
    },
  },
  component: Member,
} as Meta;

export const Default: Story = {};
