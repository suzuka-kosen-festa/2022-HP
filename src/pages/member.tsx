import type { FC } from "react";
import { Member } from "../components/templates/member";
import { rikuto, shoma, members } from "../constants/members";

const MemberPage: FC = () => <Member rikuto={rikuto} shoma={shoma} members={members} />;

export { MemberPage };
