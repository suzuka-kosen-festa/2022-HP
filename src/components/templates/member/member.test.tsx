import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./member.stories";

const { Default } = composeStories(stories);

const options = {
  name: "member page",
};

describe("(components) templates/member", () => {
  test("to be templates", () => {
    const { container } = render(<Default />);
    expect(container).toBeTemplate();
  });
  test("to be [role=main]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("main", options)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
