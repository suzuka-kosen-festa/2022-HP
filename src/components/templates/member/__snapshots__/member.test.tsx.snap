// Vitest Snapshot v1

exports[`(components) templates/member > take snap shot 1`] = `
<div>
  <main
    aria-label="member page"
    class="css-10jk6q1"
  >
    <div
      class="css-v1klec-Member"
    >
      <img
        alt="ink"
        class="css-mhf7w5-Title"
        height="580"
        loading="eager"
        src="/statics/Ink3_blue.webp"
        width="592"
      />
      <h1
        class="css-1ueto1"
      >
        Members
      </h1>
    </div>
    <div
      class="css-wkpw2c"
    >
      <h2
        class="css-1sz2wzf-Member"
      >
        第57回鈴鹿高専祭 Re:ROADの実行委員をご紹介します！
      </h2>
      <div
        class="css-14jner2-Member"
      >
        <figure>
          <img
            alt="Rikuto Kuwahara"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rikuto_kuwahara.webp"
            width="450"
          />
          <figcaption
            class="css-fyk2nh-MemberCard"
          >
            <p
              class="css-1cv9gjo"
            >
              LEADER
            </p>
            <p
              class="css-uqq0xe"
            >
              Rikuto Kuwahara
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Shoma Kobayashi"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/shoma_kobayashi.webp"
            width="450"
          />
          <figcaption
            class="css-fyk2nh-MemberCard"
          >
            <p
              class="css-1cv9gjo"
            >
              SUB LEADER
            </p>
            <p
              class="css-uqq0xe"
            >
              Shoma Kobayashi
            </p>
          </figcaption>
        </figure>
      </div>
      <div
        class="css-1gbr933-Member"
      >
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
        <figure>
          <img
            alt="Rintaro Itokawa"
            class="css-5ressk-MemberCard"
            height="300"
            loading="lazy"
            src="/statics/members/rintaro_itokawa.webp"
            width="450"
          />
          <figcaption
            class="css-1mlw4fg-MemberCard"
          >
            <p
              class="css-uqq0xe"
            >
              Rintaro Itokawa
            </p>
          </figcaption>
        </figure>
      </div>
    </div>
  </main>
</div>
`;
